symphony-bdk-core/src/test/java/com/symphony/bdk/core/activity/parsing/SlashCommandPatternTest.java

```diff
@@ -0,0 +1,314 @@
+
+    final List<CommandToken> tokens = pattern.getTokens();
+    assertEquals(1, tokens.size());
+    assertTrue(tokens.get(0) instanceof TypedArgumentToken<?>);
+
+    V4Message message = buildMessage("<span class=\"entity\" data-entity-id=\"0\">$mycashtag</span>",
+        "{\"0\":{\"id\":[{\"type\":\"org.symphonyoss.fin.security.id.ticker\",\"value\":\"mycashtag\"}],\"type\":\"org.symphonyoss.fin.security\"}}");
+
+    final MatchResult matchResult = pattern.getMatchResult(message);
+    assertTrue(matchResult.isMatching());
+    assertEquals(1, matchResult.getArguments().size());
+
+    final Object resultArg = matchResult.getArguments().get(argName);
+    assertTrue(resultArg instanceof Cashtag);
+
+    final Cashtag cashtag = (Cashtag) resultArg;
+    assertEquals("$mycashtag", cashtag.getText());
+    assertEquals("mycashtag", cashtag.getValue());
+  }
+
+  @Test
+  void oneHashtag() {
+    String argName = "myarg";
+    SlashCommandPattern pattern = new SlashCommandPattern("{#" + argName + "}");
+
+    final List<CommandToken> tokens = pattern.getTokens();
+    assertEquals(1, tokens.size());
+    assertTrue(tokens.get(0) instanceof TypedArgumentToken<?>);
+
+    V4Message message = buildMessage("<span class=\"entity\" data-entity-id=\"0\">#myhashtag</span>",
+        "{\"0\":{\"id\":[{\"type\":\"org.symphonyoss.taxonomy.hashtag\",\"value\":\"myhashtag\"}],\"type\":\"org.symphonyoss.taxonomy\"}}");
+
+    final MatchResult matchResult = pattern.getMatchResult(message);
+    assertTrue(matchResult.isMatching());
+    assertEquals(1, matchResult.getArguments().size());
+
+    final Object resultArg = matchResult.getArguments().get(argName);
+    assertTrue(resultArg instanceof Hashtag);
+
+    final Hashtag cashtag = (Hashtag) resultArg;
+    assertEquals("#myhashtag", cashtag.getText());
+    assertEquals("myhashtag", cashtag.getValue());
+  }
+
+  @Test
+  void oneStaticAndOneArgumentGluedShouldBeInvalid() {
+    assertThrows(SlashCommandSyntaxException.class, () -> new SlashCommandPattern("/command{arg}"));
+  }
+
+  @Test
+  void twoGluedArgumentsShouldBeInvalid() {
+    assertThrows(SlashCommandSyntaxException.class, () -> new SlashCommandPattern("{arg1}{arg2}"));
+  }
+
+  private MatchResult getMatchResult(SlashCommandPattern pattern, String textContent) {
+    return pattern.getMatchResult(buildMessage(textContent));
+  }
+
+  private V4Message buildMessage(String textContent) {
+    return new V4Message().message("<div data-format=\"PresentationML\" data-version=\"2.0\" class=\"wysiwyg\"><p>" + textContent + "</p></div>");
+  }
+
+  private V4Message buildMessage(String textContent, String data) {
+    return buildMessage(textContent).data(data);
+  }
+
+  private void assertIsRegexToken(String expectedRegex, CommandToken actualToken) {
+    assertTrue(actualToken instanceof RegexCommandToken);
+    assertEquals(expectedRegex, ((RegexCommandToken) actualToken).getRegexPattern().pattern());
```

Comment:
in tests I would recommend using assertj for better error messages (and readibility sometime)
not for this change maybe but in the future